restructuredtext_lint/cli.py

```diff
@@ -37,8 +37,21 @@ def _main(filepaths, format=DEFAULT_FORMAT, stream=sys.stdout, encoding=None, le
           **kwargs):
     error_dicts = []
     error_occurred = False
-
-    for filepath in filepaths:
+    paths = []
+
+    for path in filepaths:
+        # Check if the given path is a file or a directory
+        if (os.path.isfile(path)):
+            paths.append(path)
+        else:
+            for file in os.listdir(path):
+                fname, fext = os.path.splitext(file)
+                if fext == '.rst':
+                    paths.append(os.path.join(path, file))
+                elif fext == '':
```

Comment:
Unfortunately, this is going to fail for directories that contain a dot in them (e.g. `foo.app`). We probably need to iterate recursively (i.e. going through the same `.isfile` process -- the easiest way to do this should be `os.walk`)

https://docs.python.org/2/library/os.html#os.walk